app/Config.hs

```diff
@@ -66,14 +74,22 @@ instance FromJSON Config where
     configCopyFiles  <- o .:? "copy_files" .!= []
     configCopyDirs   <- o .:? "copy_dirs"  .!= []
     configVcAction    <- o .:? "vc_action" .!= True
-    configRunLocally  <- o .:? "run_locally" >>= 
+    configRunLocally  <- o .:? "run_locally" >>=
       maybe (return Nothing) (fmap Just . mapM mkCmd)
+    configTargetSystem <- o .:? "target_system" .!= GNULinux
     return Config {..}
 
 instance FromJSON CopyThing where
   parseJSON = withObject "src and dest of a thing to copy" $ \o ->
     CopyThing <$> (o .: "src") <*> (o .: "dest")
 
+instance FromJSON TargetSystem where
+  parseJSON = withText "target system" $ \ts ->
+    pure $ case ts of
```

Comment:
What about using a `LambdaCase` here?
```haskell
instance FromJSON TargetSystem where
  parseJSON = withText "target system"
    $ pure . \case
      "linux"   -> GNULinux
      "windows" -> Windows
      _         -> BSD
```

I think the yaml for the configuration will be something like:
```yml
target system: windows
```
Right? Only to be clear.